### PR TITLE
Order Aggregate Root, Repository and Unit Tests

### PR DESCRIPTION
**Pull request by AlexanderLitus**

### CODE REVIEW

**Comment by AlexanderLitus** [comment]
@alexander-yevsyukov, PTAL (order AR is not 100% covered yet).

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    }
+
+    @Apply
+    private void apply(OrderReservationCompleted event) {
+        updateSeats(event.getSeatList());
+    }
+
+    @Apply
+    private void apply(OrderTotalsCalculated event) {
+        // NOP
```
Comment (alexander-yevsyukov) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java:194]: Why are we having NOP event appliers in the aggregate?
These events are either not for this aggregate, or we need to do something about them.

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+            Validator.checkNewState(newState);
+        }
+    }
+
+    private OrderTotalsCalculated buildOrderTotalsCalculated(OrderId orderId, ConferenceId conferenceId, List<SeatQuantity> seats) {
+        final OrderTotal total = pricingService.calculateTotalOrderPrice(conferenceId, seats);
+        return EventBuilder.buildOrderTotalsCalculated(orderId, total);
+    }
+
+    private static class EventBuilder {
```
Comment (alexander-yevsyukov) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java]: From what I see in this class, it looks like a Factory, not Builder.

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+        private static OrderTotalsCalculated buildOrderTotalsCalculated(OrderId orderId, OrderTotal total) {
```
Comment (alexander-yevsyukov) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java]: Having 'build' or 'create' prefix makes it harder to read: all of the methods start from the same prefix.

Why not just orderTotalsCalculated(), orderPlaced(), etc.?

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java (hunk 1 above)
Comment (AlexanderLitus) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java:194]: We have to create `apply` methods because Spine throws an exception if there is no applier for an event that command handler generated.

File: registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+            checkField(hasId, "order ID", cmd);
+        }
+
+        private static void checkSeats(List<SeatQuantity> seats, Message cmd) {
+            checkField(!seats.isEmpty(), "seats", cmd);
+        }
+
+        private static void checkField(boolean hasField, String fieldName, Message cmd) {
+            if (!hasField) {
+                throw new IllegalArgumentException("No " + fieldName + " in the command: " + cmd.getClass().getName());
```
Comment (alexander-yevsyukov) [inline registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java]: How about this wording: "The field 'fieldName' must be defined in all commands of class <class>."?

File: registration/src/test/java/org/spine3/samples/lobby/registration/conference/ConferenceProjectionShould.java [hunk 5]
```diff
@@ -81,7 +71,7 @@ public void tearDownTest() throws IOException {
 
     @Test
     public void handle_ConferenceCreated_event_and_update_state() {
-        final ConferenceCreated event = conferenceCreated();
+        final ConferenceCreated event = BuildHelper.conferenceCreated();
```
Comment (alexander-yevsyukov) [inline registration/src/test/java/org/spine3/samples/lobby/registration/conference/ConferenceProjectionShould.java]: Please think of a better name for 'BuildHelper'.

**Comment by alexander-yevsyukov** [comment]
LGTM
